crates/ruff_linter/src/rules/wemake_python_styleguide/rules/too_short_name.rs

```diff
@@ -0,0 +1,159 @@
+/// abscissa = 2
+/// ```
+///
+/// ## Options
+/// - [`wemake-python-styleguide.min-name-length`][wemake-python-styleguide.min-name-length]
+///
+/// [wemake-python-styleguide.min-name-length]: ../../settings#wemake-python-styleguide-min-name-length
+#[violation]
+pub struct TooShortName {
+    pub name: String,
+    pub is_module: bool,
+}
+
+impl Violation for TooShortName {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        let TooShortName { name, is_module } = self;
+        format!(
+            "Found too short {}: `{}`",
+            if *is_module { "module name" } else { "name" },
+            name,
+        )
+    }
+}
+
+pub(crate) struct Checkable(Identifier);
+
+impl From<&Identifier> for Checkable {
+    fn from(value: &Identifier) -> Self {
+        Self(value.clone())
+    }
+}
+
+impl From<&Parameter> for Checkable {
+    fn from(value: &Parameter) -> Self {
+        (&value.name).into()
+    }
+}
+
+impl TryFrom<&Alias> for Checkable {
+    type Error = ();
+
+    fn try_from(value: &Alias) -> Result<Self, Self::Error> {
+        match value {
+            Alias {
+                asname: Some(identifier),
+                ..
+            } => Ok(identifier.into()),
+            _ => Err(()),
+        }
+    }
+}
+
+impl TryFrom<&Expr> for Checkable {
+    type Error = ();
+
+    fn try_from(value: &Expr) -> Result<Self, Self::Error> {
+        match value {
+            Expr::Name(ExprName { id, range, .. }) => Ok(Self(Identifier::new(id, *range))),
+            Expr::Attribute(ExprAttribute { attr, .. }) => Ok(attr.into()),
+            _ => Err(()),
+        }
+    }
+}
+
+impl TryFrom<&Box<Expr>> for Checkable {
+    type Error = ();
+
+    fn try_from(value: &Box<Expr>) -> Result<Self, Self::Error> {
+        (&(**value)).try_into()
+    }
+}
+
+/// WPS111
+pub(crate) fn too_short_name(checker: &mut Checker, node: impl TryInto<Checkable>) {
+    if let Ok(Checkable(identifier)) = node.try_into() {
+        if naming::is_too_short_name(
+            identifier.as_str(),
+            checker.settings.wemake_python_styleguide.min_name_length,
+            true,
+        ) {
+            checker.diagnostics.push(Diagnostic::new(
+                TooShortName {
+                    name: identifier.to_string(),
+                    is_module: false,
+                },
+                identifier.range(),
+            ));
+        }
+    };
+}
+
+/// WPS111 (for filesystem)
+pub(crate) fn too_short_module_name(
+    path: &Path,
+    package: Option<&Path>,
+    settings: &LinterSettings,
+) -> Option<Diagnostic> {
+    if !matches!(
+        path.extension().and_then(std::ffi::OsStr::to_str),
+        Some("py" | "pyi")
+    ) {
+        return None;
+    }
+
+    if let Some(package) = package {
+        let module_name = if naming::is_module_file(path) {
+            package.file_name().unwrap().to_string_lossy()
+        } else {
+            path.file_stem().unwrap().to_string_lossy()
```

Comment:
I copied the boilerplate from https://github.com/astral-sh/ruff/blob/20e33bf5147a027e1d0d4ef28f88cfb5b3916efe/crates/ruff_linter/src/rules/pep8_naming/rules/invalid_module_name.rs#L50 and didn't think much about it, since it seems like it hasn't caused much trouble in the `pep8-naming` implementation.